debezium-connector-mongodb/src/main/java/io/debezium/connector/mongodb/JsonSerialization.java

```diff
@@ -53,16 +50,26 @@ public static interface Transformer extends Function<Document, String> {
     private final Transformer transformer;
 
     public JsonSerialization() {
-        final Encoder<Document> encoder = MongoClientSettings.getDefaultCodecRegistry().get(Document.class);
-        transformer = (doc) -> doc.toJson(COMPACT_JSON_SETTINGS, encoder);
+        transformer = (doc) -> doc.toJson(COMPACT_JSON_SETTINGS);
     }
 
-    public String getDocumentIdOplog(Document document) {
+    public String getDocumentIdOplog(BsonDocument document) {
         if (document == null) {
             return null;
         }
         // The serialized value is in format {"_": xxx} so we need to remove the starting dummy field name and closing brace
-        final String keyValue = new BasicDBObject("_", document.get(ID_FIELD_NAME)).toJson(SIMPLE_JSON_SETTINGS);
+        BsonValue value = document.get(ID_FIELD_NAME);
+        Object obj;
+        if (value.isObjectId()) {
+            obj = value.asObjectId().getValue();
+        }
+        else if (value.isInt32()) {
+            obj = value.asInt32().getValue();
+        }
+        else {
```

Comment:
Any other types to account for here? e.g., int64, because `toJson` would cover a broader set.

Reply:
Added int64 to be safe